### PR TITLE
Add hibernate dialect

### PR DESCRIPTION
**Pull request by aibolit**
This is essentially a copy of https://github.com/gwenn/sqlite-dialect/blob/master/src/main/java/org/hibernate/dialect/SQLiteDialect.java (unlicense)
Except, in order to make DDL work, I had to change the IdentityColumnSupport to be the default IdentityColumnSupportImpl because comdb2 doesn't currently support auto-increment indentity columns.

```
./gradlew install
```

___
I am currently making use of this in https://github.com/aibolit/Comdb2Test
It requires a local comdb2 named 'testdb'
```
./gradlew bootRun
```
to see the sql output from hibernate. It will create the tables and insert some rows.

The classes are in grails-app/domain
The entities are created at grails-app/init/BootStrap.groovy

### CODE REVIEW

**Comment by riverszhang89** [comment]
Thank you for the patch! I'm a bit concerned about the copyright, although the repository is unlicensed. @kpfleming Could you please comment on this?

**Comment by kpfleming** [comment]
The author of that code has indicated that they do not wish to retain copyright on it; while it's not ideal to do it that way (since there are places in the world where copyright ownership cannot be relinquished) and they should have chosen a suitable statement like CC-0, as long as the new file retains this notice and suitable links/references to the source of the file and the author, it should be fine for inclusion.

**Comment by aibolit** [comment]
Added a reference at the top of the file to the original commit.

File: cdb2hibernatedialect/src/main/java/org/hibernate/dialect/Comdb2Dialect.java [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+  /* not case insensitive for unicode characters by default (ICU extension needed)
+	public boolean supportsCaseInsensitiveLike() {
+    return true;
+  }
+  */
+
+	@Override
+	public boolean doesReadCommittedCauseWritersToBlockReaders() {
+		// TODO Validate (WAL mode...)
+		return true;
```
Comment (riverszhang89) [inline cdb2hibernatedialect/src/main/java/org/hibernate/dialect/Comdb2Dialect.java]: Should be false.

File: cdb2hibernatedialect/src/main/java/org/hibernate/dialect/Comdb2Dialect.java [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+	}
+
+	@Override
+	public boolean doesRepeatableReadCauseReadersToBlockWriters() {
+		return true;
```
Comment (riverszhang89) [inline cdb2hibernatedialect/src/main/java/org/hibernate/dialect/Comdb2Dialect.java]: Ditto.

File: cdb2hibernatedialect/src/main/java/org/hibernate/dialect/Comdb2Dialect.java [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+		//registerColumnType(Types.DOUBLE, "double");
+		registerColumnType( Types.DECIMAL, "decimal" );
+		registerColumnType( Types.CHAR, "char" );
+		registerColumnType( Types.LONGVARCHAR, "longvarchar" );
+		registerColumnType( Types.TIMESTAMP, "datetime" );
+		registerColumnType( Types.BINARY, "blob" );
+		registerColumnType( Types.VARBINARY, "blob" );
+		registerColumnType( Types.LONGVARBINARY, "blob" );
+
+		registerFunction( "concat", new VarArgsSQLFunction( StandardBasicTypes.STRING, "", "||", "" ) );
```
Comment (riverszhang89) [inline cdb2hibernatedialect/src/main/java/org/hibernate/dialect/Comdb2Dialect.java]: Comdb2 support a lot more functions than the list.

File: cdb2hibernatedialect/src/main/java/org/hibernate/dialect/Comdb2Dialect.java [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+        registerColumnType(Types.LONGVARBINARY, "blob");
+        registerColumnType(Types.BIGINT, "longlong");
+
+        registerFunction("concat", new VarArgsSQLFunction(StandardBasicTypes.STRING, "", "||", ""));
+        registerFunction("mod", new SQLFunctionTemplate(StandardBasicTypes.INTEGER, "?1 % ?2"));
+        registerFunction("quote", new StandardSQLFunction("quote", StandardBasicTypes.STRING));
+        registerFunction("random", new NoArgSQLFunction("random", StandardBasicTypes.INTEGER));
+        registerFunction("round", new StandardSQLFunction("round"));
+        registerFunction("substr", new StandardSQLFunction("substr", StandardBasicTypes.STRING));
+        registerFunction("trim", new AbstractAnsiTrimEmulationFunction() {
```
Comment (riverszhang89) [inline cdb2hibernatedialect/src/main/java/org/hibernate/dialect/Comdb2Dialect.java]: I can complete the function list. Would you mind if I push a few commits to your branch?

Comment (aibolit) [reply cdb2hibernatedialect/src/main/java/org/hibernate/dialect/Comdb2Dialect.java]: yes; thank you. I have added you as a collaborator

**Comment by aibolit** [comment]
Looks good to me. The gradle wrapper wasn't working so I updated it with 
`$ gradle wrapper --gradle-version 4.2`

**Comment by riverszhang89** [comment]
Thanks @aibolit !

**Pull request merged**